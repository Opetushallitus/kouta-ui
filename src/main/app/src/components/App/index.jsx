import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import MainPage from '../MainPage';

const App = ({ store, theme }) => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <MainPage />
      </ThemeProvider>
    </Provider>
  );
};

export default App;