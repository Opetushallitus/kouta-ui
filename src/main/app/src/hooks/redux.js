import { useMemo } from 'react';
import { bindActionCreators } from 'redux';
import { useDispatch } from 'react-redux';
import { isArray } from 'lodash';

export function useActions(actions, deps = []) {
  const dispatch = useDispatch();
  return useMemo(() => {
    if (isArray(actions)) {
      return actions.map(a => bindActionCreators(a, dispatch));
    }
    return bindActionCreators(actions, dispatch);
  }, [actions, dispatch]);
}
