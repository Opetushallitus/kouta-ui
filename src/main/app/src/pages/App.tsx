import React, { Suspense } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { PersistGate } from 'redux-persist/integration/react';
import { I18nextProvider } from 'react-i18next';

import FullSpin from '#/src/components/FullSpin';
import GlobalStyle from '#/src/components/GlobalStyle';
import HttpContext from '#/src/contexts/HttpClientContext';
import UrlContext from '#/src/contexts/UrlContext';
import VirkailijaRaamit from '#/src/components/VirkailijaRaamit';
import UserGate from '#/src/pages/UserGate';
import HttpErrorNotifier from '#/src/components/HttpErrorNotifier';
import ErrorBoundaryNotifier from '#/src/components/ErrorBoundaryNotifier';
import Routes from './Routes';

const App = ({
  store,
  theme,
  httpClient,
  urls,
  history,
  localization,
  persistor,
}) => {
  return (
    <Provider store={store}>
      <I18nextProvider i18n={localization}>
        <ThemeProvider theme={theme}>
          <PersistGate
            persistor={persistor}
            loading={<FullSpin size="large" />}
          >
            <HttpContext.Provider value={httpClient}>
              <UrlContext.Provider value={urls}>
                <GlobalStyle />
                <Suspense fallback={<FullSpin size="large" />}>
                  <ErrorBoundaryNotifier>
                    <VirkailijaRaamit />
                    <UserGate fallback={<FullSpin size="large" />}>
                      <HttpErrorNotifier />
                      <Routes history={history} />
                    </UserGate>
                  </ErrorBoundaryNotifier>
                </Suspense>
              </UrlContext.Provider>
            </HttpContext.Provider>
          </PersistGate>
        </ThemeProvider>
      </I18nextProvider>
    </Provider>
  );
};

export default App;
