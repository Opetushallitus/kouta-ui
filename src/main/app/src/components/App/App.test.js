import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';

import App from './index';

const store = createStore((state = 0, action) => state);

test('Renders correctly', () => {
  expect(shallow(<App store={store} theme={{}} />)).toMatchSnapshot();
});
