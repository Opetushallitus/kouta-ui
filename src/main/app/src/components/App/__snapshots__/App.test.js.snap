// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ThemeProvider
    theme={Object {}}
  >
    <MainPage />
  </ThemeProvider>
</Provider>
`;
